### DISCUSSION TITLE
Last 2.XXX Version Before Project Singleton Removal

### DISCUSSION THREAD

**Discussion by pkenjora** [google/starthinker#61] [General]
This is the last version before starthinker.util.project singleton is removed.  To use a 2.XXX version:

$ python -m pip install git+https://github.com/django/django.git@[tag version]

To migrate to version 3.XXX:

1. Add (config, task) to all task entry points.
2. Add (config, ...) to all util entry points.
3. Remove all starthinker.util.project imports.
4. Re-download any generated assets like airflow modules, colabs, pip installs...

All other changes are compatible.  Recipes and UI have not changed in functionality or interface.

<hr /><em>This discussion was created from the release <a href='https://github.com/google/starthinker/releases/tag/'>Last 2.XXX Version Before Project Singleton Removal</a>.</em>

**Comment by gkulkarni2020** [comment]
Hi Team,
We are trying to deploy this on GCP using a restricted service account (non-editor). Is there any comprehensive list of all IAM Roles a service account needs to have for a successful deployment/upgrades?
